app/scripts/modules/netflix/canary/canaryConfigSelector.directive.js

```diff
@@ -0,0 +1,57 @@
+'use strict';
+
+let angular = require('angular');
+
+module.exports = angular
+  .module('spinnaker.netfilx.widgits.canaryConfigSelector.directive', [
+    require('./canary.read.service'),
+    require('../../core/utils/lodash')
+  ])
+  .directive('canaryConfigSelector', () => {
+    return {
+      scope: {},
+      bindToController: {
+        model: '=' ,
+        applicationName: '='
+      },
+      templateUrl: require('./canaryConfigSelector.directive.html'),
+      controllerAs: 'vm',
+      controller: function($scope, canaryReadService, _) {
+        let vm = this;
+
+        vm.selectorChanged = (model) => {
+          if (model) {
+            vm.model = model;
+            vm.model.successfulScore = model.canarySuccessCriteria.canaryResultScore;
+            vm.model.unhealthyScore = model.canaryHealthCheckHandler.minimumCanaryResultScore;
+            vm.model.resultStrategy = model.combinedCanaryResultStrategy;
+          }
+        };
+
+        vm.isNameInConfigList = (selectedName) => {
+          return getSelected(vm.canaryConfigs, selectedName).length ? true : false;
+        };
+
+        let setOptions = (results) => {
+          vm.canaryConfigs = results;
+          return vm.canaryConfigs;
+        };
+
+        let getSelected = _.curry((canaryConfigs = [], canaryName) => {
+          return canaryConfigs.filter((config) => config.name === canaryName);
+        });
```

Comment:
In this instance there would be little difference in the result.  What the curry buys you is the ability to partially apply a generic function to create a more specific function.   What I should have done was:

```
let getSelected = _.curry(listOfObj, property, finder) =>  {
  return listOfObj.filter((obj) => obj[property] === finder);
}
```
Then I can create a more specific function:
```
let findAllByNameThatMatch = getSelected(canaryConfigs, 'name');
```

And execute it later: 
```
let found = findAllByNameThatMatch('my-name-to-find');
```